http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/LeoPubsubMessageSubscriber.scala

```diff
@@ -269,6 +271,46 @@ class LeoPubsubMessageSubscriber[F[_]: Async: Timer: ContextShift: Concurrent](
       runtimeConfig <- RuntimeConfigQueries.getRuntimeConfig(runtime.runtimeConfigId).transaction
       _ <- runtimeConfig.cloudService.interpreter.startRuntime(StartRuntimeParams(runtime, now))
     } yield ()
+
+  private[monitor] def handleUpdateRuntimeMessage(msg: UpdateRuntimeMessage, now: Instant)(
+    implicit traceId: ApplicativeAsk[F, TraceId]
+  ): F[Unit] =
+    for {
+      runtimeOpt <- clusterQuery.getClusterById(msg.runtimeId).transaction
+      runtime <- runtimeOpt.fold(
+        Async[F].raiseError[Runtime](PubsubHandleMessageError.ClusterNotFound(msg.runtimeId, msg))
+      )(Async[F].pure)
+      runtimeConfig <- RuntimeConfigQueries.getRuntimeConfig(runtime.runtimeConfigId).transaction
+
+      // We assume all validation has already happened in RuntimeServiceInterp
+      // Resize the cluster
+      _ <- if (msg.newNumWorkers.isDefined || msg.newNumPreemptibles.isDefined)
+        runtimeConfig.cloudService.interpreter
+          .resizeCluster(ResizeClusterParams(runtime, msg.newNumWorkers, msg.newNumPreemptibles))
+      else Async[F].unit
+
+      // Update the disk size
+      _ <- msg.newDiskSize.traverse_ { d =>
+        runtimeConfig.cloudService.interpreter.updateDiskSize(UpdateDiskSizeParams(runtime, d))
+      }
+      // Update the machine type
+      // If it's a stop-update transition, persist a followup record and transition the runtime to Stopping
+      _ <- msg.newMachineType.traverse_ { m =>
+        if (msg.stopUpdateMachineType) {
+          val followupDetails = RuntimeFollowupDetails(msg.runtimeId, RuntimeStatus.Stopped)
+          for {
+            _ <- dbRef.inTransaction(
+              clusterQuery.setToStopping(msg.runtimeId, now) >> followupQuery
+                .save(followupDetails, Some(m))
+            )
+            _ <- runtimeConfig.cloudService.interpreter
+              .stopRuntime(StopRuntimeParams(RuntimeAndRuntimeConfig(runtime, runtimeConfig), now))
+          } yield ()
+        } else {
+          runtimeConfig.cloudService.interpreter.updateMachineType(UpdateMachineTypeParams(runtime, m, now))
+        }
+      }
+    } yield ()
```

Comment:
Do we need to add/update `isUpdating` flag?

Reply:
Yeah… that PR isn't merged yet, will probably need to coordinate with @akarukappadath about who goes first. (I'm happy to rebase this PR and incorporate that flag.)